File: backend/src/main/java/com/acronexus/service/StudentLoginHistoryService.java
package com.acronexus.service;

import java.util.List;
import java.util.Map;
import java.util.UUID;

public interface StudentLoginHistoryService {

    /** Record a successful student login */
    void recordLogin(UUID studentUserId);

    /** Get classes accessible to the logged-in user (HOD or Coordinator) */
    List<Map<String, Object>> getAccessibleClasses(UUID userId);

    /** Get login summary for students in a given class or ALL classes */
    List<Map<String, Object>> getClassLoginSummary(String classId, UUID requesterId);

    /** Get complete login history for a specific student */
    Map<String, Object> getStudentLoginHistory(UUID studentId, UUID requesterId);
}

File: frontend/src/services/studentLoginHistoryService.ts
import api from './api';

export const studentLoginHistoryService = {
  /** Fetch classes accessible to the current user */
  getAccessibleClasses: async () => {
    const response = await api.get('/student-login-history/classes');
    return response.data;
  },

  /** Fetch login summary for students in a given class */
  getClassLoginSummary: async (classId: string) => {
    const response = await api.get(`/student-login-history/class/${classId}/summary`);
    return response.data;
  },

  /** Fetch complete login history for an individual student */
  getStudentLoginHistory: async (studentId: string) => {
    const response = await api.get(`/student-login-history/student/${studentId}`);
    return response.data;
  },
};
